pkg/cleanup/cleanup.go

```diff
@@ -0,0 +1,52 @@
+package cleanup
+
+import (
+	"github.com/containous/maesh/pkg/k8s"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Cleaner is an interface for the cleanup methods.
+type Cleaner interface {
+	CleanShadowServices() error
+}
+
+// Ensure the Prepare fits the Preparer interface
+var _ Cleaner = (*Cleanup)(nil)
+
+// Cleanup holds the clients for the various resource controllers.
+type Cleanup struct {
+	client k8s.Client
+	log    logrus.FieldLogger
+}
+
+// NewCleanup returns an initialized cleanup object.
+func NewCleanup(log logrus.FieldLogger, client k8s.Client) Cleaner {
+	return &Cleanup{
+		client: client,
+		log:    log,
+	}
+}
+
+// CleanShadowServices deletes all shadow services from the cluster.
+func (c *Cleanup) CleanShadowServices() error {
+	serviceList, err := c.client.GetKubernetesClient().CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{
+		LabelSelector: "app=maesh",
```

Comment:
Are you sure that only shadow services will be returned?
I'm thinking that this could have side effects with the `maesh-controller` service introduced in #525.

Reply:
You are correct, we should use a new label for shadow services to prevent this.

Reply:
So if we use a new label, it will not work for previous versions of maesh, right?

Reply:
I was going to add an additional label. You are correct, the cleanup would not work on older versions without the label.

Reply:
The other option would be to have a more complex service filter that would ignore the controller and API services created by helm.

This would also be backwards compatible.

@kevinpollet which method do you think you would want to see? (new label vs more involved service search)

Reply:
I agree with what @kevinpollet said. Additionally, could we specify the maesh namespace instead of `metav1.NamespaceAll`?

Reply:
I think a new label would be good. Of course, being backwards compatible is preferred but we don't have a clean up as of now. So only, for new meshes the new feature will work. It's not like we're breaking something, it's like "it will not work".

Maybe we could even reconcile that and add the new labels to existing shadow services? @dtomcej